cmd/woodpecker-autoscaler/main.go

```diff
@@ -25,6 +26,13 @@ func setupProvider(ctx *cli.Context, config *config.Config) (engine.Provider, er
 	// Enable it again when the issue is fixed.
 	// case "linode":
 	// 	return linode.New(ctx, config)
+	case "scaleway":
+		scwCfg, err := scaleway.FromCLI(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		return scaleway.New(scwCfg, config)
```

Comment:
Was there a specific reason to use another structure to do the config setup for this provider? I would prefer to keep the setup for providers equal. Everything done by ` scaleway.FromCLI(ctx)` should be handled by `scaleway.New` similar to e.g. https://github.com/woodpecker-ci/autoscaler/blob/main/providers/hetznercloud/provider.go#L43

Reply:
I think I initially decoupled the parsing from CLI and the config structure to allow using declarative file (e.g. a yaml to config the autoscaler) instead of the CLI. But it seems like we won't support configuration from file so I can remove that indeed.

Reply:
If you have some time to change it that would be great. If not we can also merge the current state and change it later.